### ISSUE TITLE
Invalid formatstring don't produce compiler error

### ISSUE THREAD

**Issue by hirschenberger** [rust-lang/rust#40729]
This is related to #34341 

The formatstring `{:#>b}` does compile although the docs say the syntax is:

```
format_spec := [[fill]align][sign]['#']['0'][width]['.' precision][type]
```

The string `{:>#b}` would be correct for the above example.
I think more syntax-tests are required.

**Comment by hirschenberger** [comment]
Sorry, I was wrong, the `#` is the fill character before the alignment and so ` {:#>#8b}` works as expected.
